extension/memorylimiterextension/config.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package memorylimiterextension // import "go.opentelemetry.io/collector/extension/memorylimiterextension"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/memorylimiter"
+)
+
+// MemoryLimiterExtension is an Extension control memory usage of OpenTelemetry Collector.
+// Common usage is to check memory usage during connection and can reject or accept the connection based on the
+// configured memoryLimiter settings
+type MemoryLimiterExtension interface {
+	// MustRefuse returns if the caller should deny because memory has reached it's configured limits
+	MustRefuse() bool
+}
+
+type memoryLimiterExtension struct {
+	memLimiter *memorylimiter.MemoryLimiter
+}
+
+// MemoryLimitation defines memory limiter path for the receiver
+type MemoryLimitation struct {
+	// MemoryLimiterID specifies the name of the memory limiter extension
+	MemoryLimiterID component.ID `mapstructure:"memory_limiter"`
+}
+
+type Config = memorylimiter.Config
+
+// GetMemoryLimiter attempts to find a memory limiter extension in the extension list.
+// If a memory limiter extension is not found, an error is returned.
+func (m *MemoryLimitation) GetMemoryLimiter(extensions map[component.ID]component.Component) (MemoryLimiterExtension, error) {
+	if ext, found := extensions[m.MemoryLimiterID]; found {
+		if ml, ok := ext.(MemoryLimiterExtension); ok {
+			return ml, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to resolve Memory Limiter %q", m.MemoryLimiterID)
+}
```

Comment:
Let's not introduce this until we apply this to the receivers. I'm not sure this is the right approach. I don't think we should apply memory limiters implicitly. Receivers should probably explicitly define usage of a memory limiter extension in the config. Still to be discussed

Reply:
removed